File: coroutine-py/basics01.py
import asyncio


def main():
    asyncio.run(async_main())


async def async_main():
    async with asyncio.TaskGroup() as tg:
        tg.create_task(do_world())
        print('Hello')


async def do_world():
    await asyncio.sleep(1)
    print('World!')


if __name__ == "__main__":
    main()

File: asyncawait-ts/basics01.ts
import {sleep} from "./lib";

function main() {
    (async () => {
        const p = (async () => {
            await sleep(1000);
            console.log('World!');
        })();
        console.log('Hello');
        await p;
    })();
    // cannot block
}

main();
